addons/website_slides/models/slide_slide.py

```diff
@@ -335,58 +361,116 @@ def _compute_user_membership_id(self):
+            if google_drive_url:
+                match = re.match(self.GOOGLE_DRIVE_DOCUMENT_ID_REGEX, google_drive_url)
+                if match and len(match.groups()) == 2:
+                    google_drive_id = match.group(2)
+
+            if google_drive_id and slide.slide_type == 'video':
+                slide.document_google_drive_id = False
+                slide.video_google_drive_id = google_drive_id
+            elif google_drive_id and slide.slide_type in ['document', 'infographic']:
+                slide.document_google_drive_id = google_drive_id
+                slide.video_google_drive_id = False
+            else:
+                slide.document_google_drive_id = False
+                slide.video_google_drive_id = False
+
+    @api.onchange('document_url', 'video_url')
     def _on_change_url(self):
+        """ Keeping a 'onchange' because we want this behavior for the frontend.
+        Changing the document / video external URL will populate some metadata on the form view.
+        The slide metadata are also fetched in create / write overrides to ensure consistency. """
+
         self.ensure_one()
-        if self.url:
-            res = self._parse_document_url(self.url)
-            if res.get('error'):
-                raise UserError(res.get('error'))
-            values = res['values']
-            if not values.get('document_id'):
-                raise UserError(_('Please enter valid Youtube or Google Doc URL'))
-            for key, value in values.items():
-                self[key] = value
-
-    @api.onchange('datas')
-    def _on_change_datas(self):
-        """ For PDFs, we assume that it takes 5 minutes to read a page.
-            If the selected file is not a PDF, it is an image (You can
-            only upload PDF or Image file) then the slide_type is changed
-            into infographic and the uploaded dataS is transfered to the
-            image field. (It avoids the infinite loading in PDF viewer)"""
-        if self.datas:
-            data = base64.b64decode(self.datas)
-            if data.startswith(b'%PDF-'):
-                pdf = PyPDF2.PdfFileReader(io.BytesIO(data), overwriteWarnings=False, strict=False)
-                self.completion_time = (5 * len(pdf.pages)) / 60
-            else:
-                self.slide_type = 'infographic'
-                self.image_1920 = self.datas
-                self.datas = None
+        if self.document_url or self.video_url:
+            slide_metadata = self._fetch_external_metadata()
+            if slide_metadata:
+                self.update(slide_metadata)
+
+    @api.onchange('document_data_pdf')
+    def _on_change_document_data_pdf(self):
+        if self.slide_type == 'document' and self.source_type == 'local_file' and self.document_data_pdf:
+            completion_time = self._get_completion_time_pdf(base64.b64decode(self.document_data_pdf))
+            if completion_time:
+                self.completion_time = completion_time
```

Comment:
Shouldn't it be computed / editable ? based on either datas for pdf, youtube information, ... ?